src/framework/MOM_string_functions.F90

```diff
@@ -419,6 +420,41 @@ function slasher(dir)
   endif
 end function slasher
 
+!> append a string (substring) to another string (string_in) and return the
+!! concatenated string (string_out)
+function append_substring(string_in, substring) result(string_out)
+   character(len=*), intent(in) :: string_in !< input string
+   character(len=*), intent(in) :: substring !< string to append string_in
+   ! local
+   character(len=1024) :: string_out
+   character(len=1024) :: string_joined
+   integer :: string_in_length
+   integer :: substring_length
+
+   string_out = ''
+   string_joined = ''
+   string_in_length = 0
+   substring_length = 0
+
+   string_in_length = len_trim(string_in)
+
+   substring_length = len_trim(substring)
+
+   if (string_in_length > 0) then
+     if (substring_length > 0) then
+         string_joined = trim(string_in)//trim(substring)
+
+         string_out(1:len_trim(string_joined)) = trim(string_joined)
+     else
+        write(stdout,*) "WARNING: MOM_string_functions::append_substring:", &
```

Comment:
These string functions need to be pure (can not do any writing to stdout) since they might be invoked as `write(stdout,*) append_substing('a','b')`. The existing "write"s you see are in unit tests.